From f6df4c2a7fbd8c8ab95df85c712749c4555ce30d Mon Sep 17 00:00:00 2001
From: Hiroshi Hatake <hatake@clear-code.com>
Date: Wed, 19 Apr 2017 11:42:26 +0900
Subject: [PATCH] Don't show the window frames

The current EGL patch for Wayland doesn't work well when the
window frame exist. Until the issue is resolved, don't show
the frame and make the window fullscreen by default.

Signed-off-by: Takuro Ashie <ashie@homa.ne.jp>
Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>
---
 browser/base/content/browser.xul | 2 +-
 widget/gtk/nsWindow.cpp          | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/browser/base/content/browser.xul b/browser/base/content/browser.xul
index 2c74aecdf607..7a11377e3060 100644
--- a/browser/base/content/browser.xul
+++ b/browser/base/content/browser.xul
@@ -58,7 +58,7 @@
         macanimationtype="document"
         screenX="4" screenY="4"
         fullscreenbutton="true"
-        sizemode="normal"
+        sizemode="fullscreen"
         retargetdocumentfocus="urlbar"
         persist="screenX screenY width height sizemode">
 
diff --git a/widget/gtk/nsWindow.cpp b/widget/gtk/nsWindow.cpp
index 4f7f2aa871f7..5cf31b7bd0c6 100644
--- a/widget/gtk/nsWindow.cpp
+++ b/widget/gtk/nsWindow.cpp
@@ -3876,6 +3876,7 @@ nsWindow::Create(nsIWidget* aParent,
 
         // make sure this is the focus widget in the container
         gtk_widget_show(container);
+        gtk_window_set_decorated(GTK_WINDOW(mShell), FALSE);
         gtk_widget_grab_focus(container);
 
         // the drawing window
-- 
2.11.0

